src/Table/Grid.php

```diff
@@ -411,7 +411,7 @@ protected function buildColumn($name, $callback = null)
      */
     protected function buildModel($model)
     {
-        if ($this->paginate === true && method_exists($model, 'paginate')) {
+        if ($this->isEloquentModel($model) && $this->paginate) {
```

Comment:
You should be able to pass any time of data (other than eloquent or query builder) and this should just accept it as long as it has a `paginate` method.